### ISSUE TITLE
run loop several times，ax_client.get_next_trials doesn't work and “Process finished with exit code 1”

### ISSUE THREAD

**Issue by cyrilmyself** [facebook/Ax#1813]
i am testing categorical variable user Ax service API，the loop runs 10 times，the loop can not continue，

from ax.service.ax_client import AxClient
from ax.utils.measurement.synthetic_functions import branin
from ax.modelbridge.generation_strategy import GenerationStrategy, GenerationStep
from ax.modelbridge.registry import Models
from botorch.models import SingleTaskGP, MultiTaskGP
from botorch.acquisition import qNoisyExpectedImprovement
from botorch.acquisition.multi_objective.monte_carlo import qExpectedHypervolumeImprovement, qNoisyExpectedHypervolumeImprovement
from ax.models.torch.botorch_modular.surrogate import Surrogate
import torch
from botorch.test_functions.multi_objective import BraninCurrin
from ax.service.utils.instantiation import ObjectiveProperties
import numpy as np

branin_currin = BraninCurrin(negate=True).to(
    dtype=torch.double,
    device=torch.device("cuda" if torch.cuda.is_available() else "cpu"),
)

def evaluate(trial_index, parameters):

    evaluation = branin_currin(torch.tensor([parameters.get("x1"), parameters.get("x2")]))
    x = np.array(list(parameters.values()), dtype=float)
    ret = {"a": evaluation[0].item(), "b": evaluation[1].item(), "l2norm": np.sqrt(np.sum(x ** 2))}
    print(trial_index, parameters)

    return ret

def test():

    gs = GenerationStrategy(
        steps=[
            # 1. Initialization step (does not require pre-existing data and is well-suited for
            # initial sampling of the search space)
            GenerationStep(
                model=Models.SOBOL,
                num_trials=10,  # How many trials should be produced from this generation step
                min_trials_observed=10, # How many trials need to be completed to move to next model
                max_parallelism=10,  # Max parallelism for this step
                model_kwargs={"seed": 999},  # Any kwargs you want passed into the model
                model_gen_kwargs={},  # Any kwargs you want passed to `modelbridge.gen`
            ),
            # 2. Bayesian optimization step (requires data obtained from previous phase and learns
            # from all data available at the time of each new candidate generation call)
            GenerationStep(
                model=Models.BOTORCH_MODULAR, # WORKS IF COMMENTED AND REPLACED WITH Models.GPEI (comment line below, too)
              qNoisyExpectedHypervolumeImprovement},
                num_trials=-1,  # No limitation on how many trials should be produced from this step
            ),
        ]
    )

    ax_client = AxClient(generation_strategy=gs)
 
    ax_client.create_experiment(
        name="moo_experiment",
        parameters=[
            {
                "name": "x1",
                "type": "range",
                "bounds": [0.0, 1.0],
            },
            {   "name": "x2",
                "type": "choice",
                "values": [0.0, 0.05,  0.1, 0.15, 0.2, 0.25, 0.3, 0.35, 0.4, 0.45, 0,5, 0.55, 0.6, 0.65, 0.7, 0.75, 0.8, 0.85, 0.9, 0.95, 1.0, 1.05, 1.1, 1.15, 1.2],
                "value_type": 'float',
            }
        ],
        objectives={
            # `threshold` arguments are optional
            "a": ObjectiveProperties(minimize=False, threshold=branin_currin.ref_point[0]),
            "b": ObjectiveProperties(minimize=False, threshold=branin_currin.ref_point[1]) # WORKS IF COMMENTED
        },
        # parameter_constraints=["x1 + x2 <= 2.0"],  # Optional.
        outcome_constraints=["l2norm <= 1.25"],  # Optional.
        overwrite_existing_experiment=True,
        is_test=True,
    )
    

    for i in range(100):
        ret = ax_client.get_next_trials(1)
        parameters_trial_index = ret[0]

        print('get', len(parameters_trial_index))
        for trial_index, params in parameters_trial_index.items():
            ax_client.complete_trial(trial_index=trial_index, raw_data=evaluate(trial_index, params))

    pareto_optimal_parameters = ax_client.get_pareto_optimal_parameters()
    # best_parameters, metrics = ax_client.get_best_parameters()

    print(pareto_optimal_parameters)

if __name__ == '__main__':

    test()
